app/bundles/DynamicContentBundle/EventListener/DynamicContentSubscriber.php

```diff
@@ -153,14 +157,17 @@ public function onDelete(Events\DynamicContentEvent $event)
 
     public function onTokenReplacement(MauticEvents\TokenReplacementEvent $event)
     {
-        /** @var Lead $lead */
         $lead         = $event->getLead();
         $content      = $event->getContent();
         $clickthrough = $event->getClickthrough();
 
-        if ($content) {
+        if ($lead instanceof Lead && $content) {
+            /** @var Lead $lead */
```

Comment:
It is obvious that `$lead` variable must be an instance of `Lead` from the line above, right? So this comment is just stating obvious
```suggestion
```

Reply:
I am adding it to the patch. As of now, I don't recall why I have added that.